### PR TITLE
user-guide: add SSH jobs troubleshooting note

### LINKED ISSUE #4445: dvc-ssh: add section explaining how to tweak jobs parameter
### Report

In https://dvc.org/doc/user-guide/data-management/remote-storage/ssh, we should add the info from https://github.com/iterative/dvc-ssh/issues/16#issuecomment-1491774178.

### PR DESCRIPTION
**Pull request by pmrowla**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Closes #4445

### CODE REVIEW

File: content/docs/user-guide/data-management/remote-storage/ssh.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ with a remote computer, which lets you safely transfer files to and from it
 (like [`scp`]), among other features. Other operations can be used on top of
 SSH, like FTP (simple file transfer protocol) which becomes secure or [SFTP].
 
-[SSH]: https://www.ssh.com/academy/ssh
-[SFTP]: https://www.ssh.com/academy/ssh/sftp-ssh-file-transfer-protocol
+[ssh]: https://www.ssh.com/academy/ssh
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/ssh.md:16]: Q: is it done by a linter?

**Review by shcheklein — APPROVED** [review]

File: content/docs/user-guide/data-management/remote-storage/ssh.md (hunk 1 above)
Comment (pmrowla) [reply content/docs/user-guide/data-management/remote-storage/ssh.md:16]: Yes, if I run `yarn format content/docs/...` it converts all of these link entries to lowercase

**Comment by dberenbaum** [comment]
@pmrowla Are you planning on following up on https://github.com/iterative/dvc-ssh/issues/16 this sprint? If so, I don't mind waiting if we may have a different suggestion soon, like using `max_sessions` instead.

**Comment by daavoo** [comment]
Can/Should we catch the `ValueError: Can't create any SFTP connections!` and link to this?

**Comment by pmrowla** [comment]
@dberenbaum we can merge this for now, the `max_sessions` suggestion won't actually work if there's an underlying problem w/the sshfs session pool

> In practice, it may be that there is a problem with session pool implementations in sshfs where it doesn't properly handle cases where the # of active coroutines is larger than the pool size, which is why I initially suggested that we just use `fs.config.max_sessions` as the allowed maximum for `--jobs`.

the dvc-ssh/sshfs behavior needs more investigation at this point and I'm not sure how long it will take

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**